src/main/scala/com/amazon/deequ/analyzers/Distance.scala

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ * use this file except in compliance with the License. A copy of the License
+ * is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.deequ.analyzers
+
+case class CategoricalHistogramBucket(value: String, count: Long)
+
+case class CategoricalHistogram(buckets: List[CategoricalHistogramBucket])
+
+object Distance {
+
+    /** Calculate distance of numerical profiles based on KLL Sketch */
+    def calculateNumericalDistance(
+      sample1: QuantileNonSample[Double],
+      sample2: QuantileNonSample[Double],
+      calculateLinfSimple: Boolean = false)
+    : Double = {
+      val rankMap1 = sample1.getRankMap()
+      val rankMap2 = sample2.getRankMap()
+      val combinedKeys = rankMap1.keySet.union(rankMap2.keySet)
+      val n = rankMap1.valuesIterator.max.toDouble
+      val m = rankMap2.valuesIterator.max.toDouble
+      var linfSimple = 0.0
+
+      combinedKeys.foreach { key =>
+        val cdf1 = sample1.getRank(key, rankMap1) / n
+        val cdf2 = sample2.getRank(key, rankMap2) / m
+        val cdfDiff = Math.abs(cdf1 - cdf2)
+        linfSimple = Math.max(linfSimple, cdfDiff)
+      }
+      if (calculateLinfSimple) {
+        linfSimple
+      } else {
+        // This formula is based on  “Two-sample Kolmogorov–Smirnov test"
+        // Reference: https://en.m.wikipedia.org/wiki/Kolmogorov%E2%80%93Smirnov_test
+        val linfRobust = Math.max(0.0, linfSimple - 1.8 * Math.sqrt((n + m) / (n * m)))
```

Comment:
If I understand things correctly, then the "robust" computation somehow conducts a hypothesis test, right? It would be great if you could elaborate a bit more on that. I think we should also rename the `calculateLinfSimple` parameter to reflect this.

Reply:
Yes you are absolutely right, the "robust" formula conducts a hypothesis test and the formula is based on null hypothesis.  If we have enough samples from two distributions, we could simply calculate ${linfSimple}, but when there aren't enough samples, ${linfSimple} could be large not because the distributions are different but because we don't have enough observations. That's why we make it statistically robust by subtracting 1.8 * Math.sqrt((n + m) / (n * m)), where n and m are numbers of samples in these two distributions. Therefore I suggest to change the variable name from "calculateLinfSimple" to "enoughSamples" to reflect this.